File: testcases/src/main/java/inheritance/one/BaseAbstract.java
package inheritance.one;

public abstract class BaseAbstract implements ICRUD{

    @Override
    public String createREST(String transferObject) {
        return create(transferObject);
    }

    @Override
    public String deleteREST( String id) {
        return delete(id);
    }






    private String delete(String id) {
        return id;
    }

    private String create(String transferObject) {
        return transferObject;
    }


}

File: testcases_kt/src/main/kotlin/inheritance/one/BaseAbstract.kt
package inheritance.one

abstract class BaseAbstract : ICRUD {
    override fun createREST(transferObject: String): String {
        return create(transferObject)
    }

    override fun deleteREST(id: String): String {
        return delete(id)
    }

    private fun delete(id: String): String {
        return id
    }

    private fun create(transferObject: String): String {
        return transferObject
    }
}
